File: scrabble-game/src/main/java/com/aryzko/scrabblegame/application/model/MoveResult.java
package com.aryzko.scrabblegame.application.model;

import lombok.Builder;
import lombok.Singular;
import lombok.Value;

import java.util.List;

@Value
@Builder
public class MoveResult {
    String actualPlayerId;
    @Singular
    List<PlayerMove> playerMoves;
    GameState gameState;

    @Value
    @Builder
    public static class PlayerMove {
        String playerId;
        @Singular
        List<Tile> tiles;
        String word;
        Integer movePoints;
        Integer allPoints;
    }

    @Value
    @Builder
    public static class GameState {
        State state;
        String winnerId;
    }

    public enum State {
        NOT_STARTED,
        STARTED,
        FINISHED
    }
}

File: scrabble-ui/src/app/clients/game-manager/model/move-result.ts
import {Tile} from "./tile";

export interface MoveResult {
  actualPlayerId: string;
  playerMoves: PlayerMove[];
  gameState: GameState;
}

export interface PlayerMove {
  playerId: string;
  tiles: Tile[];
  word: string;
  movePoints: number;
  allPoints: number;
}

export interface GameState {
  state: string;
  winnerId: string;
}
